### PR TITLE
Update to latest base, ledger and consensus

### PR DESCRIPTION
**Pull request by dcoutts**

### CODE REVIEW

File: cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs [hunk 1]
```diff
@@ -56,8 +57,9 @@ import           Ouroboros.Consensus.Util.Condense (Condense(..))
 --
 
 instance Condense (OneEraHash xs) where
-    condense = condense . Crypto.UnsafeHash . getOneEraHash
-
+    condense = condense . unsafeHashFromBytes . getOneEraHash
+      where
+        unsafeHashFromBytes = Crypto.UnsafeHash . SBS.toShort
```
Comment (mrBliss) [inline cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs:62]: You can use `toRawHash` for this.

Comment (dcoutts) [reply cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs]: thanks, done.

File: cardano-node/src/Cardano/Node/Types.hs [hunk 2]
```diff
@@ -214,9 +214,11 @@ instance FromJSON NodeConfiguration where
       parseHardForkProtocol v = do
         npcTestShelleyHardForkAtEpoch   <- v .:? "TestShelleyHardForkAtEpoch"
         npcTestShelleyHardForkAtVersion <- v .:? "TestShelleyHardForkAtVersion"
+        npcShelleyHardForkNotBeforeEpoch <- v .:? "ShelleyHardForkNotBeforeEpoch"
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Node/Types.hs:217]: Should the default configuration file for mainnet define `ShelleyHardForkNotBeforeEpoch` to be 190, or in fact 208?

Comment (dcoutts) [reply cardano-node/src/Cardano/Node/Types.hs:217]: We'll put it into the actual config file, so it's explicit.

Comment (mrBliss) [reply cardano-node/src/Cardano/Node/Types.hs:217]: That's what I meant.

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**